pkg/domain/infra/tunnel/containers.go

```diff
@@ -92,7 +92,35 @@ func (ic *ContainerEngine) ContainerUnpause(ctx context.Context, namesOrIds []st
 
 func (ic *ContainerEngine) ContainerStop(ctx context.Context, namesOrIds []string, opts entities.StopOptions) ([]*entities.StopReport, error) {
 	reports := []*entities.StopReport{}
-	ctrs, rawInputs, err := getContainersAndInputByContext(ic.ClientCtx, opts.All, opts.Ignore, namesOrIds)
+
+	containersNamesOrIds := namesOrIds
+	if len(opts.Filters) > 0 {
+		containersNamesOrIds = []string{}
+		options := new(containers.ListOptions).WithFilters(opts.Filters).WithAll(true)
+		candidates, listErr := containers.List(ic.ClientCtx, options)
+		if listErr != nil {
+			return nil, listErr
+		}
+		for _, candidate := range candidates {
+			if opts.All {
+				containersNamesOrIds = append(containersNamesOrIds, candidate.ID)
+				continue
+			}
+			for _, nameOrID := range namesOrIds {
+				if nameOrID == candidate.ID {
+					containersNamesOrIds = append(containersNamesOrIds, nameOrID)
+					continue
+				}
+				for _, containerName := range candidate.Names {
+					if containerName == nameOrID {
+						containersNamesOrIds = append(containersNamesOrIds, nameOrID)
+						continue
+					}
+				}
+			}
+		}
+	}
+	ctrs, rawInputs, err := getContainersAndInputByContext(ic.ClientCtx, opts.All, opts.Ignore, containersNamesOrIds)
```

Comment:
I think we should call ContainerList here instead to avoid duplicating logic and code.